[lint] fixes to mypy linter

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68460
* __->__ #68479

I thought I landed this already, but:
- Don't run one mypy instance per file, run one per config
- Do the same for flake8
- Properly handle stub files

suo committed:
[lint] fixes to mypy linter

I thought I landed this already, but:
- Don't run one mypy instance per file, run one per config
- Properly handle stub files

[ghstack-poisoned]

suo committed:
Update on "[lint] fixes to mypy linter"


I thought I landed this already, but:
- Don't run one mypy instance per file, run one per config
- Properly handle stub files

[ghstack-poisoned]

suo committed:
Update on "[lint] fixes to mypy linter"


I thought I landed this already, but:
- Don't run one mypy instance per file, run one per config
- Properly handle stub files

[ghstack-poisoned]

suo committed:
Update on "[lint] fixes to mypy linter"


I thought I landed this already, but:
- Don't run one mypy instance per file, run one per config
- Properly handle stub files

[ghstack-poisoned]

suo commented:
@janeyx99 resurrecting this stack as it was basically the last step before being able to enable lintrunner in CI

suo committed:
Update on "[lint] fixes to mypy linter"


I thought I landed this already, but:
- Don't run one mypy instance per file, run one per config
- Do the same for flake8
- Properly handle stub files

[ghstack-poisoned]

janeyx99 reviewed (approved):
Looks reasonable to me

suo committed:
Update on "[lint] fixes to mypy linter"


I thought I landed this already, but:
- Don't run one mypy instance per file, run one per config
- Do the same for flake8
- Properly handle stub files

[ghstack-poisoned]

suo commented:
@pytorchbot merge this
